### PR TITLE
use username_to_user_id

### PR DESCRIPTION
**Pull request by benrudolph**
@dimagi/scale-team this uses a function that maps a username to a couch user id and then aggressively caches it. hopefully this'll provide short term relief while we work on a longer term solution. thanks @emord for the suggestion

cc: @calellowitz

### CODE REVIEW

File: corehq/apps/users/models.py [hunk 1]
```diff
@@ -1216,6 +1214,7 @@ def clear_quickcache_for_user(self):
 
         self.get_by_username.clear(self.__class__, self.username)
         self.get_by_user_id.clear(self.__class__, self.user_id)
+        username_to_user_id.clear(self.username)
```
Comment (snopoke) [inline corehq/apps/users/models.py]: does this get run on bulk upload?

Comment (benrudolph) [inline corehq/apps/users/models.py]: like the `.clear`? no, but i don't think we have to worry about that too much. the cache returns incorrectly when a user deletes a mobile worker and then creates a new one with the same username

Comment (snopoke) [inline corehq/apps/users/models.py]: I was thinking about adding users in a bulk restore but I guess it's quite unlikely that a user will try to auth before they've been created.

Comment (dannyroberts) [inline corehq/apps/users/models.py]: There's also no reason we couldn't call `user.clear_quickcache_for_user()` for each user during bulk upload if we wanted to make sure that bulk uploads don't create an inconsistent cache situation.

**Pull request merged**